### PR TITLE
fix: defined button hover effects

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
### Description of work
- Defines button hover effects

### Testing Link(s)
- [ ] Navigate to the [Component Story](https://deploy-preview-XX--dev-component-library-twig.netlify.app/?path=/story/atoms-button--button) preview

### Functional Review Steps
- [ ] Verify the buttons each transition as expected

### Design Review
- [ ] Verify the designs match the [Figma Designs](https://www.figma.com/file/OVfmCW5s1gI7xnrM8ibX2y/UI-Kit-Alpha-%5BYaleSites%5D)

### Accessibility Review
- [ ] Verify the component meets Accessibility requirements

### CODE REVIEW

**Comment by ahitson** [comment]
- 200 does feel a bit fast on the Wipe, can we go 250? 
- Does the rise option have a drop shadow?
- Should click actions be here as well?
- Please add some space between each section so that the headings look like they are a part of what follows.

**Comment by ModulesUnraveled** [comment]
@ahitson I updated the swipe duration, added the drop-shadow, and defined `active` (when the mouse is down) styles. Let me know if there's anything else!

**Review by miketullo95 — COMMENTED** [review]
hello! after looking at the button wipe animation, I still feel like it's a little too fast and jarring. Could we experiment with the speed and see what 400 looks like (I think I'm looking at 250 right now).

File: components/01-atoms/buttons/_buttons.scss [hunk 1]
```diff
@@ -29,6 +34,15 @@ $button-outline-weights: (
     outline-offset: var(--size-thickness-2);
   }
 
+  &:active {
+    // These are !important to override any hover styles defined by more
+    // specific selectors in the 'Hover styles' section below.
+    background-color: var(--color-gray-700) !important;
```
[the remark this answers is not in the document]
Comment (ModulesUnraveled) [reply components/01-atoms/buttons/_buttons.scss]: I updated the comments here according to our zoom conversation

**Comment by ModulesUnraveled** [comment]
@miketullo95 @ahitson I've update the "wipe" version to use a 400ms transition duration. Let me know if that's better

File: components/01-atoms/buttons/_buttons.scss [hunk 2]
```diff
@@ -64,19 +69,62 @@ $button-outline-weights: (
...
+      box-shadow: inset -6.5em 0 0 0 var(--color-button-bg);
+
+      &:hover {
+        box-shadow: none;
+      }
+    }
+  }
+
+  &:active {
+    // Don't ever use !important!!!
```
Comment (callinmullaney) [inline components/01-atoms/buttons/_buttons.scss:108]: 👍

File: components/01-atoms/links/_link.scss [hunk 3]
```diff
@@ -1,6 +1,9 @@
 // @TODO: These custom property settings seem like they shouldn't need to live
```
Comment (callinmullaney) [inline components/01-atoms/links/_link.scss:1]: Per out Tech Debt standards can you create ticket in Jira with these details? Also, add an @link comment here linking it back to the ticket. https://www.notion.so/fourkitchens/Technical-Debt-820b9fda626b4b98bac36f910561767f

**Review by callinmullaney — APPROVED** [review]
@ModulesUnraveled Thanks for exhausting all avenues in regards to the `!important` - While I wish we found a better way I think your experimenting showed a rare use case for `!important`. I don't think this should be considered tech debt and the comment seems verbose enough for future devs.

See my other comment in regards to the tech debt you identified and make sure that gets updated before merging. Approved!

**Comment by kara-franco** [comment]
Quick questions I had after our Zoom convo yesterday re: use of !important on active states

- What will be the consequence when a dev would like to create a custom style and change the active state styling? 
- For dark mode/theme the button active color may be different, how will we handle that?

**Review by ahitson — APPROVED** [review]

**Review by miketullo95 — APPROVED** [review]
looks great! The new animation for wipe looks good!

File: components/01-atoms/links/_link.scss (hunk 3 above)
Comment (ModulesUnraveled) [reply components/01-atoms/links/_link.scss:2]: Done

**Comment by ModulesUnraveled** [comment]
@kara-franco I don't think your first question is a concern. This is the design system, and any/every version of button styles should be accounted for here. If there needs to be a new style, it should be added to the system. So there isn't really a situation where someone would "create a custom style and change the active state styling".

As far as dark mode etc. support, I updated the styles to use custom properties instead of defined colors etc. so that those can be modified to support the various color schemes or an eventual dark mode

**Pull request merged**